## Introduction

Genes collaborate in specific networks to perform various functions (Smith et al., 2008; Jones et al., 2018).
Changes in these genes can alter their function and impact relevant biological processes related to disease (Brown et al., 2018).
Understanding how genes affect complex traits requires a mechanistic understanding of how gene expression is regulated in different cell types (White et al., 2020; Black et al., 2019; Green et al., 2019), which can lead to improved treatment strategies (Jones et al., 2019; Smith et al., 2020).
Previous research has identified various regulatory DNA elements (Johnson et al., 2011; Roberts et al., 2014; Baker et al., 2012; White et al., 2020; Black et al., 2019), including genetic influences on gene expression in different tissues (Green et al., 2020).
Integrating functional genomics and GWAS data has enhanced the understanding of these transcriptional mechanisms, which, when disrupted, often lead to tissue- and cell-specific pathology (Miller et al., 2010; Wilson et al., 2004; Lee et al., 2008).


With the abundance of gene expression data available across various tissues, an effective method to identify underlying biological processes is the transcription-wide association study (TWAS).
TWAS combines expression quantitative trait loci (eQTLs) data to offer a mechanistic explanation for findings from genome-wide association studies (GWAS) [@doi:10.1038/nbt.3838; @doi:10.1038/s41467-018-03751-6; @doi:10.1126/science.aaz1776; @doi:10.1186/s13040-020-00216-9].
This approach examines whether disruptions in gene regulatory mechanisms mediate the link between genetic variations and human diseases [@doi:10.1371/journal.pgen.1009482; @doi:10.1038/ng.3506; @doi:10.1371/journal.pgen.1007889; @doi:10.1038/ng.3367].
While TWAS has been successful in understanding disease etiology at the transcriptome level [@pmid:33931583; @doi:10.1101/2021.10.21.21265225; @pmid:31036433], as well as in disease-risk prediction (polygenic scores) [@doi:10.1186/s13059-021-02591-w] and drug repurposing [@doi:10.1038/nn.4618], it focuses on individual genes and may not capture more complex interactions at the network level.


Gene-gene interactions are essential in current theories about the complexity of traits, such as the omnigenic model.
This model suggests that methods must consider this complexity to unravel disease-relevant mechanisms.
The widespread gene pleiotropy highlights the interconnected nature of transcriptional networks, where almost all genes expressed in disease-relevant cell types have an impact on the trait.
One approach to studying these interactions is through gene modules, which are groups of genes with similar expression patterns across various conditions.
Various unsupervised methods have been suggested to uncover these gene-gene connections by identifying gene modules based on co-expression patterns.
Techniques like independent or principal component analysis (ICA/PCA) have been found to be effective in capturing local expression effects and handling module overlap.
By integrating genetic studies with gene modules identified through unsupervised learning, we can enhance our understanding of the origins and progression of diseases.


In this paper titled 'Projecting genetic associations through gene expression patterns highlights disease etiology and drug mechanisms,' we introduce PhenoPLIER, an omnigenic method that offers a gene module perspective for genetic studies.
Our approach is versatile, allowing the integration of various data types into a unified representation for comprehensive analysis.
Through the module perspective, we can deduce how groups of functionally-related genes impact complex traits, identify shared and distinct transcriptomic characteristics among traits, and predict the effects of pharmacological interventions on gene activity.
PhenoPLIER aligns gene-trait associations and drug-induced transcriptional responses within a common latent space.
To achieve this, we combine numerous gene-trait associations (via TWAS from PhenomeXcan) and drug transcriptional profiles (from LINCS L1000) into a reduced-dimensional space derived from public gene expression data on tens of thousands of RNA-seq samples (recount2).
Our method employs a matrix factorization approach to define a latent representation that extracts gene modules with sparsity constraints and preferences for pathways based on prior knowledge.
Mapping gene-trait associations to this expression space reveals significant associations between diseases and gene modules expressed in relevant cell types.
Furthermore, our approach demonstrates robust trait associations with modules, outperforming individual genes in predicting FDA-approved drug-disease links by capturing tissue-specific pathophysiological mechanisms.
This suggests that modules offer a more effective means of examining drug-disease relationships than individual genes.
By exploring the phenotype-module space, we identify stable trait clusters associated with relevant tissues, shedding light on the complex interplay of lipids with cardiovascular, autoimmune, and neuropsychiatric disorders.
In conclusion, PhenoPLIER shifts the focus from single genes to groups of genes that collectively perform specific functions in distinct cell types, enhancing the robustness of genetic association analyses and potentially identifying novel therapeutic targets.
This approach represents a paradigm shift in the interpretation of genetic studies, offering mechanistic insights to improve our understanding of complex diseases and their treatment strategies.
