### LVs reveal trait clusters with shared transcriptomic properties

![
**Cluster analysis on traits using the latent gene expression representation.**
**a)** The projection of TWAS results on 3,752 traits into the latent gene expression representation is the input data to the clustering process.
A linear (PCA) and non-linear (UMAP) dimensionality reduction techniques were applied to the input data, and five different clustering algorithms processed all data versions.
These algorithms derive partitions from the data using different parameters (such as the number of clusters), leading to an ensemble of 4,428 partitions.
Then, a distance matrix is derived by counting how many times a pair of traits was grouped in different clusters across the ensemble.
Finally, a consensus function is applied to the distance matrix to generate consolidated partitions with different numbers of clusters (from 2 to $\sqrt{n}\approx$ 60).
These final solutions were represented in the clustering tree (Figure @fig:clustering:tree).
**b)** The clusters found by the consensus function were used as labels to train a decision tree classifier on the original input data, which detects the LVs that better differentiate groups of traits.
](images/clustering/clustering_design.svg "Cluster analysis on traits"){#fig:clustering:design width="100%"}


The projection of gene-trait associations into the latent space allowed us to identify clusters of traits connected by similar transcriptional processes.
To address biases in individual clustering algorithms, we developed a consensus clustering framework that merges trait partitions from different methods.
This approach, recommended for cluster analysis on biological data, involves creating diverse data versions processed with varying methods and parameters.
A consensus function then combines these versions to produce a superior solution.
Our pipeline produced 15 final consensus clustering solutions, with the number of clusters ranging from 5 to 29.
Rather than selecting a single solution with a specific number of clusters, we utilized a clustering tree to identify stable trait groups across various resolutions.
To determine which latent variables distinguished these trait groups, we trained a decision tree classifier using the clusters as labels.


![
**Clustering tree using multiple resolutions for clusters of traits.**
Each row represents a partition/grouping of the traits, and each circle is a cluster from that partition.
The number of clusters goes from 5 to 29.
Arrows indicate how traits in one cluster move across clusters from different partitions.
Most of the clusters are preserved across different resolutions, showing highly stable solutions even with independent runs of the clustering algorithm.
<!--  -->
RDW: red cell (erythrocyte) distribution width;
BMI: body mass index;
WC: waist circumference;
HC: hip circumference;
RA: rheumatoid arthritis;
SLE: systemic lupus erythematosus;
HTN: Hypertension;
IBD: inflammatory bowel disease;
SCZ: Schizophrenia;
CAD: Coronary artery disease;
AD: Alzheimer's disease;
<!--  -->
The full lists of traits in each cluster in the last five partitions of the tree (from $k=16$ to $k=29$) are in Supplementary Data 3-7.
](images/clustering/clustering_tree.svg "Clustering tree on groups of traits"){#fig:clustering:tree width="100%"}


Phenotypes were grouped into five branches, as shown in Figure 1: a "large" branch containing most traits, a branch for heel bone-densitometry measurements, a branch for hematological assays on red blood cells, a branch for physical measures and anthropometric traits, and a "complex" branch including various measurements and disease traits.
Results were consistent across different resolutions, with traits often clustered together within branches.
Traits were observed to move between clusters, primarily within the "complex" branch and from the "large" branch to the "complex" branch.
This behavior is expected due to shared genetic and environmental factors in complex diseases.
Supplementary Data 3-7 provide detailed clustering results.


![
**Cluster-specific and general transcriptional processes associated with different diseases.**
The plot shows a submatrix of $\hat{\mathbf{M}}$ for the main trait clusters at $k$=29, considering only LVs (rows) that are well-aligned with at least one pathway.
](images/clustering/global_clustermap-plain.svg "Heatmap with gene modules and traits"){#fig:clustering:heatmap width="100%"}


We identified the key latent variables (LVs) driving clusters of traits by training decision tree classifiers on the input data with each cluster at $k$=29 as labels (see Methods).
The top discriminative LVs for each cluster were determined, some aligning with known pathways while others were novel but expressed in relevant tissues.
Certain LVs were specific to particular traits, while others were associated with a broad range of phenotypes, suggesting involvement in general biological functions.
Using our regression framework, we assessed the associations of these LVs with different traits.
For instance, LVs like LV928 and LV30, closely related to early erythrocyte lineage progenitors, were predominantly expressed during early erythropoiesis stages and significantly associated with various red blood cell assays.
In contrast, LV730, expressed in thrombocytes from cancer samples, was strongly associated with platelet assays, while LV598, expressed in corneal endothelial cells, was associated with keratometry measurements.


The autoimmune and cardiovascular disease sub-branches converged at $k=10$ in Figure 1, indicating potential shared genetic influences.
LV57, predominantly expressed in T cells (Figure 2 and Table 1), showed strong associations with autoimmune disorders in PhenomeXcan (Table 2) and replicated hypothyroidism associations in eMERGE (Table 3).
Additionally, LV57 was significantly linked to deep venous thrombosis in both databases.
In contrast, LV844 was more autoimmune-specific, with associations to various diseases in PhenomeXcan (Table 1) but lacked replication in eMERGE.
LV844 exhibited broad expression across cell types, including blood, breast organoids, myeloma cells, lung fibroblasts, and various brain cell types (Figure 3 and Table 2).


The cardiovascular sub-branch showed 129 significant LV-trait associations in PhenomeXcan and 23 in eMERGE.
LV136, associated with collagen formation and muscle contraction pathways, was linked to coronary artery disease and keratometry measurements in PhenomeXcan.
In eMERGE, LV136 was associated with coronary atherosclerosis.
This LV was expressed in various cell types such as fibroblasts, mesenchymal stem cells, osteoblasts, pancreatic stellate cells, cardiomyocytes, and adipocytes.
Within the cardiovascular sub-branch, neuropsychiatric and neurodevelopmental disorders like Alzheimer's disease, schizophrenia, and ADHD were found, previously linked to the cardiovascular system and sharing risk factors like hypertension, high cholesterol, obesity, and smoking.
Alzheimer's disease was significantly associated with LV21 in PhenomeXcan, expressed strongly in soft tissue sarcomas, monocytes/macrophages (including microglia from cortex samples), and aortic valves.
LV21 was also associated with lipids and high cholesterol in PhenomeXcan and hyperlipidemia in eMERGE.
Macrophages play a crucial role in reverse cholesterol transport and atherogenesis, while lipid metabolism in microglia is important in neurodegenerative diseases.
