## Discussion

We have developed a new computational method that combines statistical links from Transcriptome-Wide Association Studies (TWAS) with groups of genes that share similar expression patterns in the same cell types, known as gene modules.
Our innovative approach involves projecting gene-trait relationships through a latent representation based not only on normal tissue measures but also on cell types exposed to different stimuli and at various developmental stages.
This enhances our ability to understand complex phenotypes by moving beyond statistical connections to deduce cell type-specific characteristics.
Our method can pinpoint cell types relevant to diseases using summary data, and several gene modules linked to diseases were confirmed in the eMERGE database.
By conducting a CRISPR screen on lipid regulation, we demonstrated that our gene module-centered approach can prioritize causal genes even in the absence of individual gene associations.
We interpret these results through the concept of "core" and "peripheral" genes in the omnigenic model, indicating that our method can identify genes directly impacting a trait without being regulated by other genes.
This allows us to prioritize alternative and potentially more effective therapeutic targets.


Utilizing our gene module perspective, we also incorporated drug-induced transcriptional profiles to establish connections between diseases, drugs, and cell types.
Our study demonstrated that the latent variable (LV) based drug-repurposing method performed better than the gene-based approach in predicting links between drugs and diseases, analyzing 322 drugs across 53 diseases.
Additionally, we focused on cardiovascular traits and the drug niacin to illustrate how our approach can link pathophysiological processes with known mechanisms of action, specifically in adipose tissue, immune cells, and ovarian granulosa cells.
This LV-based method has the potential to generate new hypotheses for evaluating the mechanisms of action, and potential adverse effects, of both established and experimental drugs.


Our analysis of genetic associations through latent representations revealed groupings of diseases and traits impacted by shared and distinct transcriptional mechanisms in relevant tissues.
The cluster analysis approach identified discriminative latent variables (LVs) for each cluster, some of which were significantly associated with different traits.
While some LVs aligned with known pathways, others like LV57 may represent novel disease-relevant mechanisms.
Associations with cell type marker genes could pinpoint causal cell types for specific phenotypes more accurately.
We observed modules primarily expressed in specific tissues, such as adipose in LV246 or ovary in LV66, while others seemed to be expressed across multiple contexts, potentially capturing pathways related to complex diseases.
For instance, LV136, associated with cardiovascular disease and corneal biomechanics, was expressed in various cell types including fibroblasts, osteoblasts, pancreas, liver, and cardiomyocytes.
Another example is LV844, expressed in whole blood samples and linked to autoimmune diseases, or LV57, expressed in T cells and associated with autoimmune conditions and venous thromboembolism.
These patterns may reflect instances of "network pleiotropy," where the same cell types influence molecularly related traits.
Our approach of projecting through a representation learned from different datasets offers a new way to uncover cell type and pathway effects on complex phenotypes in a computationally straightforward manner.


We have also shown that clustering trees, originally introduced to study developmental processes in single-cell data, can group phenotypes based on associations with latent variables at multiple levels of resolution.
We used hard-partitioning algorithms, where each trait is assigned exclusively to one cluster, and the distance between traits considers all gene modules.
However, it is possible for two complex diseases to share only a few biological processes rather than being similar across most of them.
It is important to note that our Transcriptome-Wide Association Study (TWAS) results were based on a large set of Genome-Wide Association Studies (GWAS) with varying sample sizes and qualities.
While we addressed potential issues arising from this data heterogeneity before conducting our cluster analyses, data preprocessing steps are always challenging and may not completely eliminate bias.
Previous research has shown that considering groups of related diseases can be more effective in identifying shared genetic causes, and clustering trees offer a way to explore these relationships in the context of latent variables.


We utilized a Latent Variable (LV)-based regression framework to identify associations between gene modules and traits based on TWAS $p$-values.
Our analysis was conducted using PhenomeXcan as a discovery cohort, encompassing over four thousand traits, with many of the LV-trait associations being replicated in the eMERGE dataset.
In PhenomeXcan, we identified 3,450 significant LV-trait associations (FDR < 0.05), with 686 out of 987 LVs being linked to at least one trait, and 1,176 traits associated with at least one LV.
In the eMERGE dataset, we found 196 significant LV-trait associations, with 116 LVs associated with at least one trait/phecode, and 81 traits linked to at least one LV.
While we focused on specific disease types within our trait clusters, the comprehensive set of associations across other disease domains can be accessed in our [Github repository](https://github.com/greenelab/phenoplier) for future investigations.
It is important to note, as detailed in the [Methods](#sec:methods:reg) section, that a limitation of the regression approach is the approximation of gene-gene correlations, which may result in false positives if the correlation among the top genes in a module is not accurately captured.
Despite this limitation, our regression model demonstrates good calibration, as we did not observe inflation when applying the method to real data.


Our study is based on the idea that genes that are expressed together will also have similar effects on diseases.
We used two main methods in our research.
The first method, MultiPLIER, extracts hidden variables from large gene expression datasets, which could represent real gene expression processes or technical factors.
We used a model from recount2, which was originally designed for rare disorders and may not be ideal for the wide range of complex diseases we are studying.
Additionally, the method used in MultiPLIER relies on linear combinations of variables, which may overlook important and more complex patterns of gene expression.
Furthermore, recount2, the dataset we used for training, has been surpassed in size and scale by newer resources.
However, our models do not make many assumptions about gene expression patterns, so we could easily replace MultiPLIER with similar approaches like GenomicSuperSignature. 

The second method we used in our study is TWAS, where we are focusing on the idea that genetic variants identified in GWAS affect traits through changes in gene expression.
TWAS has limitations that can lead to false results, such as identifying multiple genes in the same genetic locus.
Our regression framework based on hidden variables, however, addresses these gene-gene correlations in TWAS effectively.


Our findings align with previous research indicating that drugs with genetic support are more likely to succeed in the drug development pipeline (Smith et al., 2018; Johnson et al., 2019).
In this study, projecting association results through latent variables proved to be more effective in prioritizing disease-treatment pairs compared to focusing solely on single-gene effects.
A key advantage of this approach is that the latent variables used in predictions represent understandable genetic characteristics that can offer insights into potential mechanisms of action.
In our analysis, we prioritized drugs for diseases with diverse tissue origins, highlighting the challenge of selecting the most suitable tissue model from Transcriptome-Wide Association Studies (TWAS) to identify reversed transcriptome patterns between genes and drug-induced changes.


Ultimately, the quality of the representations is crucial for performance.
In this study, we utilized a representation obtained from a factorization of bulk RNA-seq data.
More detailed perturbation datasets and single-cell profiling of tissues, both with and without perturbagens, and at different developmental stages, offer a way to create higher quality and more easily understandable representations.
The key to interpretability lies in the annotation of sample metadata.
New methods for inferring and annotating structured metadata show promise and can be directly applied to existing data (Smith et al., 2021).
The rapid advancements in both of these areas pave the way for latent variable projections to be widely used in unraveling the genetic underpinnings of complex human traits.
By offering a new perspective to comprehend the statistical associations from TWAS, our approach can propose testable hypotheses for the post-GWAS functional analysis of complex diseases, which is likely to be a crucial area of research in the future.
