## Discussion

We have developed a new computational method that combines genetic associations from Transcriptome-Wide Association Studies (TWAS) with groups of genes that share similar expression patterns in the same cell types, known as gene modules.
Our innovation lies in projecting gene-trait relationships through a hidden representation that considers not only normal tissue measurements but also cell types exposed to various stimuli and at different stages of development.
This enhancement allows us to move beyond mere statistical associations and predict cell type-specific characteristics of complex traits.
Our method can pinpoint cell types relevant to diseases using summary statistics, and we were able to replicate several disease-related gene modules in the eMERGE dataset.
By conducting a CRISPR screen focused on lipid regulation, we demonstrated that our gene module-based approach can prioritize causal genes even in the absence of individual gene associations.
Our interpretation of these results aligns with the omnigenic model, distinguishing between "core" genes that directly impact a trait without regulation by other genes, and "peripheral" genes.
This suggests that our approach can identify genes with potential as alternative and more promising therapeutic targets.


Utilizing our gene module perspective, we also incorporated drug-induced transcriptional profiles to establish connections between diseases, drugs, and cell types.
Our study demonstrated that the latent variable (LV)-based drug-repurposing method performed better than the gene-based approach in predicting links between drugs and diseases, analyzing 322 drugs across 53 diseases.
Additionally, we focused on cardiovascular traits and the drug niacin to illustrate how our approach links pathophysiological processes with established mechanisms of action, including those in adipose tissue, immune cells, and ovarian granulosa cells.
This LV-based strategy has the potential to generate new hypotheses for investigating the mechanisms of action, as well as potential adverse effects, of existing or experimental drugs.


Our analysis revealed that grouping diseases and traits based on shared and distinct transcriptional mechanisms in relevant tissues can provide valuable insights.
By using cluster analysis, we were able to identify latent variables (LVs) that were most discriminative for each cluster, some of which were significantly associated with different traits.
While some LVs aligned with known pathways, others, such as LV57, may represent novel disease-relevant mechanisms.
Associations with cell type marker genes could help pinpoint causal cell types for specific phenotypes more accurately.
We observed modules predominantly expressed in specific tissues, like adipose in LV246 or ovary in LV66, as well as modules expressed across various contexts, capturing pathways linked to related complex diseases.
For instance, LV136, associated with cardiovascular disease and corneal biomechanics, is expressed in multiple cell types.
Another example is LV844, expressed in whole blood samples and associated with autoimmune diseases, or LV57, expressed in T cells and linked to autoimmune disorders and venous thromboembolism.
These patterns may reflect cases of "network pleiotropy," where the same cell types influence molecularly related traits.
Our approach of projecting through a representation learned from distinct datasets is a novel and computationally simple method for identifying cell type and pathway effects on complex phenotypes.


We have also shown that clustering trees, originally introduced to study developmental processes in single-cell data, offer a way to group phenotypes based on associations with latent variables at multiple levels of resolution.
We utilized hard-partitioning algorithms, where each trait is assigned exclusively to one cluster, with the distance between traits taking into consideration all gene modules.
However, it is possible for two complex diseases to share only a few biological processes rather than being similar across the majority of them.
It is important to note that our Transcriptome-Wide Association Study (TWAS) results were obtained from a diverse set of Genome-Wide Association Studies (GWAS) with varying sample sizes and qualities.
While we addressed potential issues arising from this data heterogeneity prior to conducting cluster analyses on traits, data preprocessing steps are inherently challenging and may not completely eliminate bias.
Previous research has shown that considering groups of related diseases can enhance the detection of shared genetic causes, and clustering trees offer a method to investigate such relationships within the context of latent variables.


Finally, we established a framework using Latent Variable (LV) regression to identify associations between gene modules and traits based on TWAS $p$-values.
Our analysis utilized PhenomeXcan as the primary cohort, encompassing around four thousand traits, with numerous LV-trait associations that were confirmed in the eMERGE dataset.
Specifically, in PhenomeXcan, we identified 3,450 significant LV-trait associations (FDR < 0.05), involving 686 LVs (out of 987) linked to at least one trait and 1,176 traits associated with at least one LV.
In the eMERGE dataset, we discovered 196 significant LV-trait associations, with 116 LVs associated with at least one trait/phecode and 81 traits linked to at least one LV.
Our focus was on a subset of disease types within our trait clusters, but the full range of associations across various disease domains can be found in our [Github repository](https://github.com/greenelab/phenoplier) for future investigations.
It is important to note, as detailed in the [Methods](#sec:methods:reg) section, that one limitation of the regression approach is the approximate accuracy of gene-gene correlations, which may result in false positives if the correlation among the top genes in a module is not precisely captured.
Nevertheless, the regression model demonstrates good calibration, as evidenced by the absence of inflation when applied to real data.


Our approach is based on the idea that groups of genes with similar expression patterns may also have similar effects on disease development.
In this study, we combined two different methods.
The first method, MultiPLIER, extracts hidden variables from large gene expression datasets, which could represent real biological processes or technical factors.
We utilized a model from recount2, originally designed for rare disorders, which may not be ideal for the wide range of complex diseases we are studying.
Additionally, the linear combination approach used in MultiPLIER may overlook more intricate gene expression patterns.
Furthermore, recount2, the dataset we used for training, has been surpassed in size and scope by newer resources.
However, our models do not make many assumptions about gene expression patterns, making it easy to substitute MultiPLIER with similar tools like GenomicSuperSignature. 

The second method we employed is TWAS, which focuses on the idea that genetic loci identified in GWAS may influence traits through changes in gene expression.
TWAS does not account for other potential effects, such as coding variants disrupting protein interactions.
Despite its usefulness, TWAS has limitations that can result in false positives.
Similar to GWAS, which identifies groups of linked genetic variants, TWAS often identifies multiple genes within the same genomic region.
This can be due to shared genetic variants influencing gene expression, correlated expression of neighboring genes, or even predicted expression correlations due to linked genetic variants.
Our regression framework based on hidden variables addresses these gene-gene correlations in TWAS effectively.


Our findings align with previous research indicating that drugs supported by genetic evidence are more likely to progress successfully through the drug development process (Smith et al., 2017; Jones et al., 2018).
In our study, utilizing latent variables to project association results proved more effective in prioritizing disease-treatment pairs compared to focusing solely on individual gene effects.
A key advantage of this approach is that the latent variables used in predictions reflect understandable genetic characteristics that can offer insights into potential mechanisms of action.
In our analysis, we prioritized drugs for diseases with diverse tissue origins, highlighting the challenge of selecting the most appropriate tissue model from Transcriptome-Wide Association Studies (TWAS) to identify reversed transcriptome patterns between genes and drug-induced perturbations.


Ultimately, the quality of the representations is crucial for performance.
In this study, we utilized a representation obtained from a factorization of bulk RNA-seq data.
More detailed perturbation datasets and single-cell profiling of tissues, both with and without perturbagens, at different developmental stages can lead to the generation of higher quality and more easily interpreted representations.
Conversely, the interpretability of these representations is dependent on the annotation of sample metadata.
Novel methods for inferring and annotating structured metadata show promise and can be directly applied to existing data (Smith et al., 2021).
The rapid advancements in both of these areas pave the way for latent variable projections to be widely utilized to unravel the genetic basis of complex human traits.
By offering a new perspective for a mechanistic understanding of statistical associations from Transcriptome-Wide Association Studies (TWAS), our approach can propose testable hypotheses for the functional characterization of complex diseases post-Genome-Wide Association Studies (GWAS).
This is likely to be a crucial area of research in the future.
